api/v1/routes/auth.py

```diff
@@ -228,6 +228,39 @@ async def verify_signin_token(token_schema: TokenRequest, db: Session = Depends(
 
     return response
 
+# Verify Magic Link
+@auth.post("/verify-magic-link")
+async def verify_magic_link(token_schema: Token, db: Session = Depends(get_db)):
+    user, access_token = AuthService.verify_magic_token(token_schema.access_token, db)
+
+    refresh_token = user_service.create_refresh_token(user_id=user.id)
+
+    response = success_response(
+        status_code=200,
+        message='Login successful',
+        data={
+            'access_token': access_token,
+            'token_type': 'bearer',
+            'user': jsonable_encoder(
+                user, 
+                exclude=['password', 'is_super_admin', 'is_deleted', 'is_verified', 'updated_at']
+            ),
+        }
+    )
+
+    # Add refresh token to cookies
+    response.set_cookie(
+        key="refresh_token",
+        value=refresh_token,
+        expires=timedelta(days=30),
+        httponly=True,
+        secure=True,
+        samesite="none",
+    )
+
+    return response
+
+    
```

Comment:
this is a duplicate you're adding

Reply:
Those are changes i pulled

Reply:
![image](https://github.com/user-attachments/assets/ccc86916-d84f-48c9-8402-15fdbdf0d50a)

Reply:
![image](https://github.com/user-attachments/assets/18b410a5-cbbe-4ded-b695-df8493407cc4)

Reply:
It says there "Joshua Oloton, 2 days ago"

Reply:
@Goketech  I have not touched that file